opencog/atoms/execution/ExecutionOutputLink.cc

```diff
@@ -195,6 +196,49 @@ Handle ExecutionOutputLink::do_execute(AtomSpace* as,
 #endif /* HAVE_CYTHON */
 	}
 
+	if (0 == schema.compare(0,4,"lib:", 4))
+	{
+		// Be friendly, and strip leading white-space, if any.
+		size_t pos = 4;
+		while (' ' == schema[pos]) pos++;
+
+        //Get the name of the Library and Function
+        //They should be sperated by a .
+        std::size_t dotpos = schema.find("\\");
+        if (dotpos == std::string::npos)
+        {
+            throw RuntimeException(TRACE_INFO,
+                "LibName and FunctionName not seperated by a '\\'");
+        }
+        std::string libName  = schema.substr(pos,dotpos-pos);
+        std::string funcName = schema.substr(dotpos+1);
+
+        void *libHandle;
+        void * tmp;
+        UUID (*func)(AtomSpace* ,UUID);
+
+        //Try and load the Library and the Function
+        if ( (libHandle = dlopen(libName.c_str(), RTLD_LAZY)) == NULL ||
+             (tmp       = dlsym(libHandle, funcName.c_str())) == NULL )
+        {
```

Comment:
Well this will load a given library at run time to call 1 function from it. I wanted to be able to call Haskell Code from the AtomSpace and since Haskell is usually compiled we will have to load a library at runtime and call function from that instead of parsing the code. Of course this code will work with any kind of library that is in a C-style.